packages/shared/src/features/onboarding/shared/FunnelStepper.tsx

```diff
@@ -220,34 +223,36 @@ export const FunnelStepper = ({
             showSkipButton={skip.hasTarget}
             showProgressBar={skip.hasTarget}
           />
-          <PaymentContextProvider
-            disabledEvents={[CheckoutEventNames.CHECKOUT_LOADED]}
-            successCallback={successCallback}
-          >
-            {funnel.chapters.map((chapter: FunnelChapter) => (
-              <Fragment key={chapter?.id}>
-                {chapter?.steps?.map((funnelStep: FunnelStep) => {
-                  const isActive = funnelStep?.id === step?.id;
-                  const Wrapper = isActive ? Fragment : HiddenStep;
-                  return (
-                    <Wrapper
-                      key={`${chapter?.id}-${funnelStep?.id}`}
-                      {...(!isActive && {
-                        'data-testid': `funnel-step`,
-                      })}
-                    >
-                      <FunnelStepComponent
-                        {...funnelStep}
-                        isActive={isActive}
-                        key={step.id}
-                        onTransition={onTransition}
-                      />
-                    </Wrapper>
-                  );
-                })}
-              </Fragment>
-            ))}
-          </PaymentContextProvider>
+          <FunnelPaymentPricingContext.Provider value={{ pricing }}>
```

Comment:
Just wrapped the whole thing. Other than that, nothing really changed.

Reply:
Did we look at the option to return the same and full config from funnel then? This mix and match of just pricings seems to break the whole concept of having metadata tied to pricings.

Reply:
We didn't want to add as many internal requests within the config since it might get bloated and increase boot time. I tried introducing a new experiment variant within our system to contain all the metadata relating to the funnel in our API db but after some discussion, it was deemed having another endpoint would be best for now.

Reply:
I don't know, seems like something that would branch out and again us end up having two or three or X different logics.

Reply:
This logic is only going to be used for Paddle platform as the Funnel is solely exclusive to web - hence why the coded affected is the provider used by Paddle only.

Reply:
As said in thread on slack, I don't agree with direction, but since you are more into the problem I trust you can manage ok solution.

Reply:
Thank you for the compromise. We'll make sure to closely monitor the discussed options before we move further after this implementation.